### PR TITLE
feat(fix): add fix-commit marker and context for inline QA/UAT (#150)

### LINKED ISSUE #150: fix and debug should suggest verify AND auto-route to UAT when auto_uat is enabled
**Problem**

Two gaps in the fix/debug → verify flow when `auto_uat=true`:

### Gap 1: suggest-next.sh doesn't suggest verify

The `fix` and `debug` branches in `scripts/suggest-next.sh` have no auto_uat/verify logic, unlike `execute`, `qa`, `status`, and `resume`. After a fix or debug session, the user never sees a `/vbw:verify` suggestion — even when `auto_uat=true` and `has_unverified_phases=true`.

Current behavior:
- `fix` → suggests `/vbw:qa`, `/vbw:vibe`
- `debug` → suggests `/vbw:fix`, `/vbw:vibe`

The user only gets routed to verify when they next run `/vbw:vibe` (priority 7 in `vibe.md`).

### Gap 2: fix/debug don't auto-route to UAT

When `auto_uat=true`, the execute flow chains into verify via vibe.md's priority table. But fix and debug commands end without any UAT routing — they just show suggestions. If auto_uat is enabled, fix/debug should be able to trigger UAT directly after completion (or at least route to it), similar to how execute → verify works.

The codebase *does* support running UAT independently via `/vbw:verify` (it auto-detects phases with SUMMARY but no UAT). The infrastructure exists — it's just not wired into the fix/debug exit path.

**Current codebase state:**
- `commands/fix.md` — spawns Dev agent in turbo mode, commits, runs `suggest-next.sh fix`, done
- `commands/debug.md` — spawns Debugger agent(s), reports findings, runs `suggest-next.sh debug`, done
- Neither command reads phase-detect state or checks `auto_uat`/`has_unverified_phases`
- `commands/verify.md` works independently and auto-detects unverified phases
- `vibe.md` priority 7 catches `auto_uat + has_unverified_phases` on next `/vbw:vibe` call

**Proposed solution**

#### Part 1: suggest-next.sh suggestions

Add verify suggestions to the `fix` and `debug` branches when auto_uat conditions are met:

```bash
fix)
    suggest "/vbw:qa -- Verify the fix"
    # NEW: if auto_uat enabled and unverified phases exist, suggest UAT
    if [ "$cfg_auto_uat" = true ] && [ "$has_unverified_phases" = true ]; then
      suggest "/vbw:verify -- Walk through changes before continuing"
    fi
    suggest "/vbw:vibe -- Continue building"
    ;;

debug)
    suggest "/vbw:fix -- Apply the fix"
    # NEW: same auto_uat verify suggestion
    if [ "$cfg_auto_uat" = true ] && [ "$has_unverified_phases" = true ]; then
      suggest "/vbw:verify -- Walk through changes before continuing"
    fi
    suggest "/vbw:vibe -- Continue building"
    ;;
```

The `current_uat_issues_phase` guard (skip verify during active remediation) should also apply here.

#### Part 2: Auto-route to UAT from fix/debug commands

When `auto_uat=true` and `has_unverified_phases=true`, fix.md and debug.md should auto-trigger UAT after completion — not just suggest it. Options:

**Option A (lighter touch):** At the end of fix.md/debug.md, after the Dev/Debugger agent completes, check phase-detect output. If `auto_uat=true && has_unverified_phases=true`, display a message like "Auto-UAT: routing to verification" and invoke `/vbw:verify` automatically.

**Option B (heavier):** Add a post-completion hook that detects fix/debug completion and routes to verify. More infrastructure but cleaner separation of concerns.

Option A is simpler and consistent with how execute → verify chaining works through vibe.md's priority table.

**Alternatives considered**

1. **Leave as-is** — rely on `/vbw:vibe` priority 7. Downside: breaks the auto_uat promise. The user expects automatic verification gating; having to manually run `/vbw:vibe` after every fix/debug defeats the purpose.
2. **Only add suggestions, no auto-routing** — addresses Gap 1 but not Gap 2. The user still needs to manually choose verify.
3. **Only add to `fix` branch** — debug is more exploratory so UAT feels less natural there. But if the debug session resulted in code changes (via `/vbw:fix` from the debug suggestion), the same auto_uat logic should apply.

**Additional context**

Related PR: #149 (fix/auto-uat-mid-milestone)
Discovered during QA Round 4 of PR #149.
Files: `scripts/suggest-next.sh` lines ~613-621 (fix branch), ~623-626 (debug branch), `commands/fix.md`, `commands/debug.md`

### PR DESCRIPTION
**Pull request by dpearson2699**
Fixes #150

## What

Adds fix-commit marker infrastructure and context compilation scripts that bridge `/fix` success to downstream QA/UAT. After a successful fix commit, a marker file is written to `.vbw-planning/.last-fix-commit` containing commit metadata. `suggest-next.sh` conditionally offers a `/vbw:verify` UAT suggestion when `auto_uat=true` and a fresh marker exists.

**Files modified:**
- `scripts/write-fix-marker.sh` (NEW) — writes `.vbw-planning/.last-fix-commit` with commit hash, message, timestamp, description, and changed files
- `scripts/compile-fix-commit-context.sh` (NEW) — reads the marker and produces compact QA/UAT context (`fix_context=available` or `fix_context=empty`)
- `commands/fix.md` (MODIFIED) — calls `write-fix-marker.sh` in both success paths (no-issues and with-discovered-issues) before `suggest-next.sh`
- `scripts/suggest-next.sh` (MODIFIED) — fix branch now conditionally suggests `/vbw:verify -- Run UAT on the fix` when `auto_uat=true`, marker exists, and marker is fresh (<24h)
- `tests/write-fix-marker.bats` (NEW) — 8 BATS tests covering marker creation, root commits, graceful degradation, and fix.md ordering contract
- `tests/compile-fix-commit-context.bats` (NEW) — 7 BATS tests covering available/empty output, UAT mode, staleness, round-trip
- `tests/suggest-next-fix-uat.bats` (NEW) — 6 BATS tests covering UAT suggestion gating (auto_uat, marker presence, staleness)

## Why

`/fix` had no post-commit QA/UAT bridge — after a successful fix, the user had to manually invoke `/vbw:qa` or `/vbw:verify`. This adds Phase 1 infrastructure (marker + context + conditional suggestion) for the inline QA/UAT lifecycle that #405 will complete. The marker pattern follows the existing `write-debug-session.sh` / `compile-debug-session-context.sh` pattern.

## How

- **write-fix-marker.sh**: Reads `git rev-parse --short HEAD`, `git log --format`, and `git diff-tree --root` to capture commit metadata. Writes key=value marker. Always exits 0 (non-blocking helper).
- **compile-fix-commit-context.sh**: Reads marker, checks 24h staleness via `stat` (platform-aware macOS/Linux), runs `git show --stat` for change summary. Two modes (qa/uat) with different titles.
- **fix.md**: Both success paths call `write-fix-marker.sh .vbw-planning` silently before `suggest-next.sh fix`.
- **suggest-next.sh**: Fix branch checks `cfg_auto_uat=true` AND marker exists AND inline `stat`-based 24h freshness check before adding UAT suggestion.

## Acceptance Criteria Verification

1. **Marker written with commit, message, timestamp, description, files** — satisfied by `write-fix-marker.sh` lines 33-50, tested in `write-fix-marker.bats` tests 1-3, 6-7
2. **compile-fix-commit-context.sh reads marker → fix_context=available/empty** — satisfied by `compile-fix-commit-context.sh` lines 59-129, tested in `compile-fix-commit-context.bats` tests 1-4
3. **Stale markers (>24h) → fix_context=empty** — satisfied by `compile-fix-commit-context.sh` lines 34-49, tested in `compile-fix-commit-context.bats` test 5
4. **suggest-next.sh suggests UAT when auto_uat=true AND fresh marker exists** — satisfied by `suggest-next.sh` lines 762-780, tested in `suggest-next-fix-uat.bats` test 4
5. **Does NOT suggest UAT when auto_uat=false, marker missing, or marker stale** — satisfied by same gating logic, tested in `suggest-next-fix-uat.bats` tests 2, 3, 5
6. **fix.md calls write-fix-marker.sh in both success paths before suggest-next.sh** — satisfied by `fix.md` lines 94-95 and 118-119, tested in `write-fix-marker.bats` test 8 (ordering contract)
7. **Scripts exit 0 gracefully** — satisfied by explicit `exit 0` paths, tested in `write-fix-marker.bats` tests 4-5
8. **BATS tests cover all changes** — 21 new BATS tests across 3 files, including staleness, root-commit, and ordering contract
9. **testing/run-all.sh green** — 2874 passed, 0 failed (lint 1/1, contracts 39/39)

## Testing

- 3 new BATS test files (21 tests total)
- Full suite: 2874 BATS passed, 39 contract checks passed, lint clean
- Known flaky: `phase-detect.bats` test tracked in #414 (pre-existing, not caused by this change)

## QA Summary

- **Primary QA (Claude):** 3 rounds, all clean (0 findings)
- **Cross-model QA (GPT-5.4):** 3 rounds
  - Round 1: 3 medium findings — F-01 (staleness check missing in suggest-next.sh), F-02 (contradictory STOP in fix.md), F-03 (observation: flaky test #414, pre-existing)
  - Round 2: 1 medium + 1 low — F-01 (root-commit files empty, `--root` flag fix), F-02 (root-commit test gap)
  - Round 3: 1 low — F-01 (contract test for marker ordering in fix.md)
  - All findings fixed
- **Copilot PR review:** pending

### CODE REVIEW

File: commands/fix.md [hunk 1]
```diff
@@ -113,7 +114,8 @@ Config: Pre-injected by SessionStart hook.
     This is **display-only**. Do NOT edit STATE.md, do NOT add todos, do NOT
     invoke /vbw:todo, and do NOT enter an interactive loop. The user decides
     whether to track these. If no discovered issues: omit the section entirely.
-    After displaying discovered issues, STOP. Do not take further action.
+    After displaying discovered issues, **STOP. Do not take further action** on discovered issues (no auto-fix, no auto-track, no investigation)—just display them.
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply commands/fix.md:117]: Not applying this suggestion. The `STOP` keyword is required by a contract test (`discovered-issues-surfacing.bats`) that verifies all discovered-issues sections include a STOP directive matching the pattern `STOP.*Do not take further action`. The current wording — `**STOP. Do not take further action** on discovered issues (no auto-fix, no auto-track, no investigation)—just display them.` — already scopes the STOP to discovered issues only and does not prevent the subsequent marker write + suggest-next execution, which appear after the STOP block.

File: scripts/write-fix-marker.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# Require git
+if ! command -v git &>/dev/null; then
+  exit 0
+fi
+
+# Read HEAD commit info
+commit_hash=$(git rev-parse --short HEAD 2>/dev/null) || exit 0
+commit_message=$(git log --format='%s' -1 2>/dev/null) || exit 0
+commit_timestamp=$(git log --format='%aI' -1 2>/dev/null) || exit 0
+changed_files=$(git diff-tree --root --no-commit-id --name-only -r HEAD 2>/dev/null) || exit 0
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/write-fix-marker.sh]: Fixed in d3727a0. Changed to `git rev-parse HEAD` (full 40-char hash) for unambiguous downstream lookups via `git show`.

File: scripts/write-fix-marker.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+# write-fix-marker.sh — Persist the latest fix commit for inline QA/UAT.
+#
+# Writes .last-fix-commit marker to the planning directory so that
+# suggest-next.sh and verify.md can detect recent fix work and offer
+# UAT without requiring PLAN/SUMMARY artifacts.
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/write-fix-marker.sh:6]: Fixed in d3727a0. Updated header comment to document that the `files=` field is a multi-line continuation (one path per line, read until next `key=` or EOF).

File: scripts/compile-fix-commit-context.sh [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  now=$(date +%s 2>/dev/null) || now=0
+  if [ "$marker_mtime" -gt 0 ] && [ "$now" -gt 0 ]; then
+    age=$(( now - marker_mtime ))
+    if [ "$age" -gt 86400 ]; then
+      echo "fix_context=empty"
+      exit 0
+    fi
+  fi
+fi
+
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/compile-fix-commit-context.sh]: Fixed in d3727a0. Replaced OSTYPE branching with portable `stat -c '%Y' || stat -f '%m' || echo 0` fallback and fail-closed: if mtime or now is ≤ 0, output `fix_context=empty` immediately.

File: tests/write-fix-marker.bats [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+# ── graceful degradation ─────────────────────────────────
+
+@test "exits 0 when planning dir does not exist" {
+  run bash "$SCRIPTS_DIR/write-fix-marker.sh" "/nonexistent/path"
+  [ "$status" -eq 0 ]
+}
+
+@test "exits 0 with default planning dir when called with no args" {
+  # No .vbw-planning in cwd, so it exits gracefully
+  cd /tmp
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/write-fix-marker.bats]: Fixed in d3727a0. Replaced `cd /tmp` with test-owned directory `$TEST_TEMP_DIR/no-default-planning` for proper isolation.

File: scripts/suggest-next.sh [hunk 6]
```diff
@@ -760,6 +760,23 @@ case "$CMD" in
...
+          _marker_mtime=$(stat -c '%Y' "$PLANNING_DIR/.last-fix-commit" 2>/dev/null) || _marker_mtime=0
+        fi
+        _now=$(date +%s 2>/dev/null) || _now=0
+        if [ "$_marker_mtime" -gt 0 ] && [ "$_now" -gt 0 ]; then
+          _marker_age=$(( _now - _marker_mtime ))
+          [ "$_marker_age" -gt 86400 ] && _marker_stale=true
+        fi
+        if [ "$_marker_stale" = false ]; then
+          suggest "/vbw:verify -- Run UAT on the fix"
+        fi
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/suggest-next.sh:774]: Fixed in d3727a0. Changed staleness check to fail-closed: `_marker_stale=true` by default, only set false when both `stat` and `date` succeed and age ≤ 24h. Also replaced OSTYPE branching with portable `stat -c '%Y' || stat -f '%m' || echo 0` fallback.

File: scripts/suggest-next.sh [hunk 7]
```diff
@@ -760,6 +760,19 @@ case "$CMD" in
...
     if [ "$_fix_debug_handled" = false ]; then
       suggest "/vbw:vibe -- Verify the fix"
+      if [ "$cfg_auto_uat" = true ] && [ -f "$PLANNING_DIR/.last-fix-commit" ]; then
+        # Only suggest UAT if marker is fresh (<24h); fail-closed on stat/date errors
+        _marker_stale=true
+        _marker_mtime=$(stat -c '%Y' "$PLANNING_DIR/.last-fix-commit" 2>/dev/null || stat -f '%m' "$PLANNING_DIR/.last-fix-commit" 2>/dev/null || echo 0)
+        _now=$(date +%s 2>/dev/null || echo 0)
+        if [ "$_marker_mtime" -gt 0 ] && [ "$_now" -gt 0 ]; then
+          _marker_age=$(( _now - _marker_mtime ))
+          [ "$_marker_age" -le 86400 ] && _marker_stale=false
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/suggest-next.sh]: Fixed in 4e370be. Added `_marker_age -ge 0` guard so future-timestamped markers are treated as stale. Also added a BATS test for this edge case.

File: tests/write-fix-marker.bats [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+  # Each marker call must precede a suggest-next call
+  for m_line in $marker_lines; do
+    local found_after=false
+    for s_line in $suggest_lines; do
+      if [ "$s_line" -gt "$m_line" ]; then
+        found_after=true
+        break
+      fi
+    done
+    [ "$found_after" = true ]
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/write-fix-marker.bats]: Fixed in 4e370be. Tightened the ordering contract to verify each marker call has a matching `suggest-next.sh` call in the same success block (between it and the next marker), rather than allowing any later suggest-next anywhere in the file to satisfy the condition.

File: scripts/write-fix-marker.sh [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+# Write marker file (overwrite any existing marker)
+marker_file="$PLANNING_DIR/.last-fix-commit"
+{
+  printf 'commit=%s\n' "$commit_hash"
+  printf 'message=%s\n' "$commit_message"
+  printf 'timestamp=%s\n' "$commit_timestamp"
+  printf 'description=%s\n' "$DESCRIPTION"
+  printf 'files=%s\n' "$changed_files"
+} > "$marker_file" 2>/dev/null || true
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/write-fix-marker.sh:57]: Fixed in 4e370be. Added `tr '\n\r' '  '` sanitization to collapse `DESCRIPTION` to a single line before writing, preventing marker format corruption. Also added a BATS test verifying newlines are stripped.

File: scripts/compile-fix-commit-context.sh [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+marker_mtime=$(stat -c '%Y' "$marker_file" 2>/dev/null || stat -f '%m' "$marker_file" 2>/dev/null || echo 0)
+now=$(date +%s 2>/dev/null || echo 0)
+
+if [ "$marker_mtime" -le 0 ] || [ "$now" -le 0 ]; then
+  echo "fix_context=empty"
+  exit 0
+fi
+
+age=$(( now - marker_mtime ))
+if [ "$age" -gt 86400 ]; then
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/compile-fix-commit-context.sh]: Fixed in 4e370be. Added `age -lt 0` guard so future-timestamped markers are treated as stale (exits with `fix_context=empty`). Also added a BATS test for this edge case.
